SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp

```diff
@@ -292,6 +292,51 @@ UClass* USpatialClassInfoManager::GetClassByComponentId(Worker_ComponentId Compo
 	return nullptr;
 }
 
+uint32 USpatialClassInfoManager::GetComponentIdForClass(const UClass& Class) const
+{
+	const FString ClassPath = Class.GetPathName();
+	if (const FSchemaData* SchemaData = SchemaDatabase->ClassPathToSchema.Find(ClassPath))
+	{
+		return SchemaData->SchemaComponents[SCHEMA_Data];
+	}
+	return SpatialConstants::INVALID_COMPONENT_ID;
+}
+
+TArray<Worker_ComponentId> USpatialClassInfoManager::GetComponentIdsForClass(const UClass& BaseClass, const bool bIncludeDerivedTypes /* = true */) const
```

Comment:
Doing a UClass reference looks pretty odd. Considering we use UClass* in the rest of the codebase, could we do that here as well?

Reply:
IMO a pointer should be used if it's valid for the parameter to be null. Pass by ref should be used otherwise. 

(https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-ptr-ref)

Reply:
I agree with that, which is what we do with FClassInfos. UClass& just doesn't exist anywhere else in the codebase, though I guess it's fine to start now.